### PR TITLE
feature/#42 Added exists options for BucketSettings

### PR DESCRIPTION
**Pull request by Rumpelshtinskiy**
Closes #

### Please check if the PR fulfills these requirements

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] CHANGELOG.md has been updated (for bug fixes / features / docs)

### What kind of change does this PR introduce?

(Bug fix, feature, docs update, ...)

### What was changed?

(Describe the changes)

### Related issues

(Add links to related issues)

### Does this PR introduce a breaking change?

(What changes might users need to make in their application due to this PR?)

### CODE REVIEW

File: src/main/java/store/reduct/client/ReductClient.java [hunk 1]
```diff
@@ -92,14 +98,15 @@ public Bucket createBucket(String name, BucketSettings bucketSettings)
 																									// settings as null
 																									// until invoke
 																									// read.
-		if (httpResponse.statusCode() == 200) {
+		if (httpResponse.statusCode() == 200 || Boolean.TRUE.equals(bucketSettings.getExists())) {
```
Comment (atimin) [inline src/main/java/store/reduct/client/ReductClient.java:101]: `Boolean.TRUE.equals(bucketSettings.getExists()))` is this really needed!? 

why not: 
```java
if (httpResponse.statusCode() == 200 || bucketSettings.getExists())
```

File: src/main/java/store/reduct/client/ReductClient.java [hunk 2]
```diff
@@ -56,6 +52,16 @@ public <T> HttpResponse<T> send(HttpRequest.Builder builder, HttpResponse.BodyHa
 		}
 	}
 
+	public <T> HttpResponse<T> send2(HttpRequest.Builder builder, HttpResponse.BodyHandler<T> bodyHandler) {
```
Comment (atimin) [inline src/main/java/store/reduct/client/ReductClient.java:55]: send2 doesn't help to understand the difference with send1.  Give them please some sensible names and little documentation.

File: src/main/java/store/reduct/client/ReductClient.java (hunk 1 above)
Comment (atimin) [inline src/main/java/store/reduct/client/ReductClient.java:101]: You pass all errors here including 401 and 403. You need to use the getExists() flag with the 409 error.

Boolean.TRUE.equals(bucketSettings.getExists())) is this really needed!?

```java
if (httpResponse.statusCode() == 200 || ( bucketSettings.getExists() && httpResponse.statusCode() == 409)

```

**Review by atimin — CHANGES_REQUESTED** [review]

File: src/main/java/store/reduct/client/ReductClient.java (hunk 1 above)
Comment (Rumpelshtinskiy) [reply src/main/java/store/reduct/client/ReductClient.java:101]: bucketSettings.getExists() return a boxed boolean - Boolean. All boxed base classes could be as null.
httpResponse.statusCode() == 200 || bucketSettings.getExists() will throw a NullPointerException if httpResponse.statusCode() is not equal to 200

Comment (Rumpelshtinskiy) [reply src/main/java/store/reduct/client/ReductClient.java]: Fixed

File: src/main/java/store/reduct/client/ReductClient.java (hunk 2 above)
Comment (Rumpelshtinskiy) [reply src/main/java/store/reduct/client/ReductClient.java]: Done

**Review by atimin — CHANGES_REQUESTED** [review]
All good with settings and options, but had found that this example stopped working:

```java
package store.reduct;

import store.reduct.client.ReductClient;
import store.reduct.client.config.ServerProperties;
import store.reduct.model.bucket.Bucket;
import store.reduct.model.bucket.BucketSettings;
import store.reduct.model.bucket.QuotaType;
import store.reduct.model.record.Record;

import java.net.http.HttpClient;
import java.util.Iterator;

public class QuickStart {
    public static void main(String[] args) {
         // 1. Create a ReductStore client
        ServerProperties properties = ServerProperties
                .builder()
                .url("http://localhost:8383")
                .apiToken("my-token")
                .build();

        HttpClient httpClient = HttpClient.newHttpClient();
        ReductClient client = new ReductClient(properties, httpClient);

        // 2. Get or create a bucket with 1Gb quota
        BucketSettings settings = BucketSettings.builder()
                .quotaType(QuotaType.FIFO)
                .quotaSize(1_000_000_000)
                .exists(true)
                .build();
        Bucket bucket = client.createBucket("my-bucket1", settings);

        // 3. Write some data with timestamps in the 'sensor-1' entry
        long timestamp = System.currentTimeMillis() * 1000; // in microseconds
        Record record = Record.builder()
                .entryName("sensor-1")
                .timestamp(timestamp)
                .type("application/octet-stream")
                .body("Record #1".getBytes())
                .build();

        bucket.writeRecord(record);
        record = Record.builder()
                .entryName("sensor-2")
                .timestamp(timestamp + 1000_000)
                .type("application/json")
                .length(9)
                .body("Record #2".getBytes())
                .build();
        bucket.writeRecord(record);

        // 4. Read the data back
        for (Iterator<Record> it = bucket.query("sensor-1", timestamp, timestamp + 1000_000, 60L); it.hasNext(); ) {
            Record rec = it.next();

            System.out.println("Record timestamp: " + rec.getTimestamp());
            System.out.println("Record size: " + rec.getBody().length);
        }

    }
}
```

It crashes on the second iteration in the for loop:

```
> Task :store.reduct.QuickStart.main() FAILED
Record timestamp: 1728910303396000
Record size: 9

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.8/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
5 actionable tasks: 2 executed, 3 up-to-date
Exception in thread "main" store.reduct.common.exception.ReductException: Query 1 not found and it might have expired. Check TTL in your query request. Default value 60 sec.
	at store.reduct.client.ReductClient.sendAndGetOnlySuccess(ReductClient.java:78)
	at store.reduct.model.bucket.Bucket$RecordIterator.hasNext(Bucket.java:356)
	at store.reduct.QuickStart.main(QuickStart.java:53)
```

**Comment by Rumpelshtinskiy** [comment]
> ```java
> for (Iterator<Record> it = bucket.query("sensor-1", timestamp, timestamp + 1000_000, 60L); it.hasNext(); ) {
> ```

I have found the reason. Wrong using byteBuffer.get method and client.send.
By the way, when you will test again, take a look on **bucket.query("sensor-1", timestamp, timestamp + 1000_000, 60L)**, the request demand only **sensor-1** , meanwhile was saved two records **sensor-1** and **sensor-2** .
Bucket.query will return only one record.

**Comment by Rumpelshtinskiy** [comment]
The branch was updated whith fix

**Pull request merged**